projects/storefrontapp-e2e-cypress/cypress/helpers/b2b/b2b-order-approval.ts

```diff
@@ -0,0 +1,128 @@
+import * as sampleData from '../../sample-data/b2b-order-approval';
+import {
+  approvalOrderDetail,
+  approvalOrderList,
+  approvedOrderDetails,
+  b2bApprover,
+  b2bUser,
+  pendingOrder,
+  rejectedOrderDetails,
+} from '../../sample-data/b2b-order-approval';
+import { AccountData } from '../../support/require-logged-in.commands';
+import { visitHomePage, waitForPage } from '../checkout-flow';
+
+export function visitOrderApprovalListPage() {
+  const alias = waitForPage(
+    '/my-account/approval-dashboard',
+    'approvalListPage'
+  );
+
+  cy.visit(`/my-account/approval-dashboard`);
+  cy.wait(`@${alias}`).its('status').should('eq', 200);
+}
+
+export function visitOrderApprovalDetailPage() {
+  const alias = waitForPage(
+    `/my-account/approval/${sampleData.approvalOrderDetail.code}`,
+    'approvalDetailPage'
+  );
+
+  cy.visit(`/my-account/approval/${sampleData.approvalOrderDetail.code}`);
+  cy.wait(`@${alias}`).its('status').should('eq', 200);
+}
+
+export function loginB2bUser() {
+  const user: AccountData = {
+    user: '',
+    registrationData: {
+      firstName: '',
+      lastName: '',
+      password: '12341234',
+      titleCode: 'mr',
+      email: b2bUser.uid,
+    },
+  };
+  cy.requireLoggedIn(user);
+  visitHomePage();
+}
+
+export function loginB2bApprover() {
+  const approver: AccountData = {
+    user: '',
+    registrationData: {
+      firstName: '',
+      lastName: '',
+      password: '12341234',
+      titleCode: 'mr',
+      email: b2bApprover.uid,
+    },
+  };
+  cy.requireLoggedIn(approver);
+  visitHomePage();
+}
+
+export function getPendingOrderDetails() {
```

Comment:
can we make more explicit in the name of these functions that we are stubbing the response of an xhr call?

Reply:
Done. We update the names of all functions which use a stubbed response.